### ISSUE TITLE
[Bug]: False positive on ec2_securitygroup_allow_ingress_from_internet_to_any_port

### ISSUE THREAD

**Issue by kenske** [toniblyx/prowler#2442]
### Steps to Reproduce

prowler --checks ec2_securitygroup_allow_ingress_from_internet_to_any_port --verbose

### Expected behavior

No findings

### Actual Result with Screenshots or Logs

```
Security group <redacted> (sg-abc123) has all ports open to the Internet.
```

### How did you install Prowler?

Cloning the repository from github.com (git clone)

### Environment Resource

EKS

### OS used

Amazon Linux 2

### Prowler version

3.5.3

### Pip version

23.1.2

### Context

Here is the security group in question. It has ports 80 and 443 open for ingress, and all egress traffic is allowed:

```
{
    "SecurityGroups": [
        {
            "Description": "Managed SecurityGroup for LoadBalancer",
            "GroupName": "<redacted>",
            "IpPermissions": [
                {
                    "FromPort": 80,
                    "IpProtocol": "tcp",
                    "IpRanges": [
                        {
                            "CidrIp": "0.0.0.0/0",
                            "Description": ""
                        }
                    ],
                    "Ipv6Ranges": [],
                    "PrefixListIds": [],
                    "ToPort": 80,
                    "UserIdGroupPairs": []
                },
                {
                    "FromPort": 443,
                    "IpProtocol": "tcp",
                    "IpRanges": [
                        {
                            "CidrIp": "0.0.0.0/0",
                            "Description": ""
                        }
                    ],
                    "Ipv6Ranges": [],
                    "PrefixListIds": [],
                    "ToPort": 443,
                    "UserIdGroupPairs": []
                }
            ],
            "OwnerId": "<redacted>",
            "GroupId": "sg-<redacted>",
            "IpPermissionsEgress": [
                {
                    "IpProtocol": "-1",
                    "IpRanges": [
                        {
                            "CidrIp": "0.0.0.0/0"
                        }
                    ],
                    "Ipv6Ranges": [],
                    "PrefixListIds": [],
                    "UserIdGroupPairs": []
                }
            ],
            "VpcId": "vpc-<redacted>"
        },
    ]
}
```

**Comment by MrCloudSec** [comment]
Good catch @kenske , thank you!!
We've just fixed this issue and will be available in the next release.

**Comment by dazzag24** [comment]
Hi, 
I can see this was fixed in [3.6.0](https://github.com/prowler-cloud/prowler/releases/tag/3.6.0) last year but I'm seeing the same issue in Prowler version 4.3.5.

I have a security group with ports 80 and 443 open for ingress, and all egress traffic is allowed:

```
{
    "SecurityGroups": [
        {
            "Description": "Foo",
            "GroupName": "<redacted>",
            "IpPermissions": [
                {
                    "FromPort": 80,
                    "IpProtocol": "tcp",
                    "IpRanges": [
                        {
                            "CidrIp": "0.0.0.0/0"
                        }
                    ],
                    "Ipv6Ranges": [],
                    "PrefixListIds": [],
                    "ToPort": 80,
                    "UserIdGroupPairs": []
                },
                {
                    "FromPort": 443,
                    "IpProtocol": "tcp",
                    "IpRanges": [
                        {
                            "CidrIp": "0.0.0.0/0"
                        }
                    ],
                    "Ipv6Ranges": [],
                    "PrefixListIds": [],
                    "ToPort": 443,
                    "UserIdGroupPairs": []
                }
            ],
            "OwnerId": "<redacted>",
            "GroupId": "sg-<redacted>",
            "IpPermissionsEgress": [
                {
                    "IpProtocol": "-1",
                    "IpRanges": [
                        {
                            "CidrIp": "0.0.0.0/0"
                        }
                    ],
                    "Ipv6Ranges": [],
                    "PrefixListIds": [],
                    "UserIdGroupPairs": []
                }
            ],
            "Tags": [
                {
                    "Key": "Name",
                    "Value": "<redacted>"
                }
            ],
            "VpcId": "vpc-<redacted>"
        }
    ]
}
```

which is the same as the one [above](https://github.com/prowler-cloud/prowler/issues/2442#issue-1742290958).

which results in a Prowler finding off:
```
Details

ResourceUid:
arn:aws:ec2:eu-west-2:<redacted>:security-group/sg-<redacted>

FindingUid:
prowler-aws-ec2_securitygroup_allow_ingress_from_internet_to_any_port-<redacted>-eu-west-2-sg-<redacted>

CheckId:
ec2_securitygroup_allow_ingress_from_internet_to_any_port

Type:
AwsEc2SecurityGroup

Details:
<redacted>

StatusExtended:
Security group <redacted>- (sg-<redacted>) has at least one port open to the Internet and neither its network interface type (interface) nor its network interface instance owner (<redacted>) are part of the allowed network interfaces.

Risk:
If Security groups are not properly configured the attack surface is increased.

Notes:

Provider:
aws

Recomendation:
Use a Zero Trust approach. Narrow ingress traffic as much as possible. Consider north-south as well as east-west traffic.

RecomendationUrl:
https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html

Scan Day:
2024-09-20
```
